flow360/plugins/report/report_items.py

```diff
@@ -1339,6 +1289,229 @@ def get_doc_item(self, context: ReportContext, settings: Settings = None) -> Non
+            x_data, y_data, x_label, y_label
+        )
+
+        for idx, (x_series, y_series, x_component, y_component) in enumerate(
+            zip(x_data, y_data, x_components, y_components)
+        ):
+            if isinstance(x_series, case_results.PerEntityResultCSVModel):
+                x_series.filter(include=self.include, exclude=self.exclude)
+                x_data[idx] = x_series.values[x_component]
+            if isinstance(y_series, case_results.PerEntityResultCSVModel):
+                y_series.filter(include=self.include, exclude=self.exclude)
+                y_data[idx] = y_series.values[y_component]
+
+        return x_data, y_data, x_label, y_label
+
+    def _get_background_chart(self, x_data):
+        if self.background == "geometry":
+            dimension = np.amax(x_data[0]) - np.amin(x_data[0])
+            if self.x == "x_slicing_force_distribution/X":
+                log.warning(
+                    "First case is used as a background image with dimensions matched to the extent of X data"
+                )
+                camera = Camera(
+                    position=(0, -1, 0), up=(0, 0, 1), dimension=dimension, dimension_dir="width"
+                )
+            elif self.x == "y_slicing_force_distribution/Y":
+                log.warning(
+                    "First case is used as a background image with dimensions matched to the extent of X data"
+                )
+                camera = Camera(
+                    position=(-1, 0, 0), up=(0, 0, 1), dimension=dimension, dimension_dir="width"
+                )
+            else:
+                raise ValueError(
+                    f"background={self.background} can be only used with x == x_slicing_force_distribution/X"
+                    + " OR x == y_slicing_force_distribution/Y"
+                )
+            background = Chart3D(
+                show="boundaries",
+                camera=camera,
+                fig_name="background_" + self.fig_name,
+                include=self.include,
+                exclude=self.exclude,
+            )
+            return background
+        return None
+
+    def get_background_chart3d(self, cases) -> Tuple[Chart3D, Case]:
+        """
+        Returns Chart3D for background.
+        """
+        # pylint: disable=unsubscriptable-object
+        reference_case_idx = self.select_indices[0] if self.select_indices is not None else 0
+        reference_case = cases[reference_case_idx]
+        x_data, _, _, _ = self._load_data([reference_case])
+        return self._get_background_chart(x_data), reference_case
+
+
+class NonlinearResiduals(BaseChart2D):
+    """
+    Residuals is an object for showing the solution history of nonlinear residuals.
+
+    """
+
+    section_title: Literal["Nonlinear residuals"] = Field("Nonlinear residuals", frozen=True)
+    fig_name: Literal["fig-residuals"] = Field("fig-residuals", frozen=True)
```

Comment:
I think fig_name should not be literal. If you want to create separate charts with nonlinear residual, the PNG files will overwrite themselves. I think it will break validation if you use two `NonlinearResiduals()` instances in one ReportTemplate: https://github.com/flexcompute/Flow360/blob/7ec8f3886aebba6b34bf90cd4d64f103a9d8d6ea/flow360/plugins/report/report.py#L158

I think you can just remove this line:
```suggestion
```